poh/src/poh_recorder.rs

```diff
@@ -183,6 +183,13 @@ impl TransactionRecorder {
                         starting_transaction_index: None,
                     };
                 }
+                Err(PohRecorderError::SendError(_)) => {
+                    return RecordTransactionsSummary {
+                        record_transactions_timings,
+                        result: Err(PohRecorderError::MaxHeightReached),
```

Comment:
I feel this overloads `MaxHeightReached`, as that's not the error we hit.
With that in mind, is it even necessary to handle these different errors differently here? It feels most natural to have this function return any error encountered directly - if the calling code wants to treat them differently it should.

Reply:
that makes sense, changed to just bubble up the internal error